lib/newProfile.ml

```diff
@@ -0,0 +1,200 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+type settings =
+  { file : string
+  }
+
+type accu =
+  | No
+  | File of out_channel
+
+let accu = ref No
+
+let pid = Unix.getpid()
+
+let is_profiling () = !accu <> No
+
+let f fmt = match !accu with
+  | No -> assert false
+  | File ch -> Printf.fprintf ch fmt
+
+module MiniJson = struct
+  type t =
+    | String of string
+    | Int of string (* string not int so that we can have large ints *)
+    | Record of (string * t) list
+    | Array of t list
```

Comment:
What I've done in the past in other projects when I needed JSON is use polymorphic variants that are compatible with those of yojson. For instance, here you could do:
```suggestion
  type t = [ 
    | `Intlit of string
    | `String of string
    | `Assoc of (string * t) list
    | `List of t list
  ]
```
which would be compatible with `Yojson.Safe.t` via subtyping I believe.

Maybe using a type like this (which is isomorphic to yours) would suit @ejgallego better.

Reply:
Indeed, tho I'd be still concerned about having to deploy our own utils for pretty printing, etc...

IMHO we need to have the discussion about using external deps in the project. I can understand that it is frustating to have that discussion on PRs where this issue is just tangential, and indeed it seems not a good choice to block this PR on that discussion.

However on the other hand, due to postponing the discussion, we keep adding technical debt to Coq, or limiting our options as developers, so each time we go ahead and add things like this we save some time, at the cost of maybe a lot of time in the future. Delaying also has a lot of cost. I dunno how to do best.

In this case note that we already have another JSON type in the extraction plugin, right?

Reply:
So apart from the external deps discussion that honestly I'm unsure how to have, I think @rlepigre suggestion is good actually, pretty easy to do, and means that external tools that already use `Yojson.Safe.t` can interop fine with this, so maybe it is worth tweaking it before merge.

Reply:
I'm fine with that suggestion.

I do like having control of the printing here because it means we can print 1 event / line, which means we can use grep to do a bit of manual analysis when the data gets big enough that smarter tools fail (specifically I tried jq on one of the big files from the earlier benches and it just kept allocating IIRC)

Reply:
That's fine, in a later PR we need to have the print to happen to a buffer (not a file) tho, so the LSP client can relay this info without having to do a round-trip by the FS. The buffer may get big indeed, but it is assumed that for this kind of tracing you need quite a beefy machine anyways.